File: Data Structures & Algorithms/find-duplicate-integer/submission-0.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        fast = 0
        slow = 0

        while True:
            slow = nums[slow]
            fast = nums[nums[fast]]

            if slow == fast:
                break

        fast = 0

        while True:
            slow = nums[slow]
            fast = nums[fast]

            if slow == fast:
                return slow
        return -1

File: Data Structures & Algorithms/duplicate-integer/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let array = []
        for (let i = 0; i < nums.length; i++) {
            if (array.some(item => item == nums[i])) {
                return true
            }
            array.push(nums[i])
        }
        return false
    }
}
